### ISSUE TITLE
How do I retrieve the RGB buffer data in this example

### ISSUE THREAD

**Issue by qq88797833** [robmikh/Win32CaptureSample#32]

**Comment by qq88797833** [comment]
I want to take out the RGB buffer and push it to the server. I hope you can help me

**Comment by qq88797833** [comment]
![image](https://user-images.githubusercontent.com/28855943/106348605-688baa00-6302-11eb-8cd1-dc2a870ceb75.png)


This is how blogs get RGB data but I get empty data why

**Comment by robmikh** [comment]
The first recommendation I would make is to observe all `HRESULT`s returned by D3D methods. They'll provide a hint as to why a call might be failing. In this case, `ID3D11DeviceContext::Map` is probably returning `E_INVALIDARG`. That's because you're trying to map the back buffer of the swapchain, but the swapchain is not initialized to be readable by the CPU (the `D3D11_USAGE` enumeration when creating a texture).

If you don't need to present the frames you get back locally, I would do away with the swapchain all together. It's only there to show you what the capture API returns for the sample. Take a look at [this issue](https://github.com/robmikh/Win32CaptureSample/issues/25) where someone asked a similar question.

In short, you'll need to do the following:

- Create a staging texture of the same dimensions and pixel format as the textures in the `Direct3D11CaptureFramePool`. Primarily, you'll want to set the `Usage` field to `D3D11_USAGE_STAGING` and the `CPUAccessFlags` field to `D3D11_CPU_ACCESS_READ` when creating your staging texture.
- When you get a new frame, copy the frame to the staging texture using `CopyResource`.
- Map and copy the bits of the staging texture. [This comment](https://github.com/robmikh/Win32CaptureSample/issues/25#issuecomment-718265205) has more details on that, but you have that part correct.
- Unmap the texture.

**Comment by robmikh** [comment]
Oh, one more thing. You'll want to call `GetDesc` and get the Width and Height of the texture. That way you won't need to hard-code it like you have in the snippet you posted.

Let me know if you run into any issues retrieving the bits.

**Comment by qq88797833** [comment]
thanks You're right . You're right. ComPtr<ID3D11DeviceContext > pD3DDC, does not recognize
![image](https://user-images.githubusercontent.com/28855943/106351591-43099b00-6318-11eb-82d0-e1fd4e489ba0.png)

**Comment by qq88797833** [comment]
Thanks. That's all right for now
